src/components/search.js

```diff
@@ -1,86 +1,88 @@
 /**
  * Search
+ * @flow
  */
 
 import React from 'react'
 import debounce from 'debounce'
-import Button from './ui/button'
-import UniqueID from '../mixins/uniqueId'
 import DataList from './datalist'
-import createClass from 'create-react-class'
-import { func } from 'prop-types'
-
-let Search = createClass({
-  mixins: [UniqueID],
-
-  statics: {
-    // The minimum number of characters before searching
-    THRESHOLD: 2,
-    // The minimum time between change events
-    INTERVAL: 150
-  },
-
-  propTypes: {
-    onChange: func.isRequired
-  },
-
-  getDefaultProps() {
-    return {
-      datalist: []
-    }
-  },
+import { type Record } from '../record'
 
-  getInitialState() {
-    return {
-      search: '',
-      debouncedChange: debounce(this.props.onChange, Search.INTERVAL)
-    }
-  },
+type Props = {
+  datalist: Record[],
+  onChange: string => *
+}
+
+type State = {
+  search: string
+}
+
+let uid = 0
+
+// The minimum number of characters before searching
+const THRESHOLD = 2
+
+// The minimum time between change events
+const INTERVAL = 150
+
+export default class Search extends React.Component<Props, State> {
+  id: number
+  debouncedChange: string => *
+
+  state = {
+    search: ''
+  }
+
+  constructor(props: Props) {
+    super(...arguments)
+
+    this.id = uid++
```

Comment:
would using `this.key` also suffice for a unique ID? I guess I'm not really sure what `uid` is doing.

Reply:
It's just giving me a counter that I can use to generate a unique `id` attribute for the input + label and the associated data list.

`this.key` is undefined. As far as I know, the key React uses to track a component is private.